src/wp-includes/block-template-utils.php

```diff
@@ -579,14 +582,16 @@ function _wp_build_title_and_description_for_single_post_type_block_template( $p
 		$post_title
 	);
 
-	$posts_with_same_title = get_posts(
-		array(
-			'title'       => $post_title,
-			'post_type'   => $post_type,
-			'post_status' => 'publish',
-		)
+	$args = array(
+		'title'                  => $post_title,
+		'post_type'              => $post_type,
+		'post_status'            => 'publish',
+		'update_post_meta_cache' => false,
+		'update_post_term_cache' => false,
```

Comment:
Should this have number => 2 as the test is for the existence of duplicates but doesn't indicate how many there are? Is it worth it?

It could probably do with these too:
'ignore_sticky_posts' => true,
'no_found_rows' => true,